docs/adr/adr-014-bootstrap-from-previous-peers.md

```diff
@@ -0,0 +1,474 @@
+# ADR 014: Bootstrap from previous peers
+
+## Changelog
+
+* 2023-05-04: initial draft
+* 2023-05-04: update peer selection with new simpler solution + fix language
+* 2023-10-04: propose a second approach
+* 2023-11-04: propose a third approach
+* 2023-14-04: document decision
+
+## Context
+
+Celestia node relies on a set of peer to peer protocols that assume a decentralized network, where nodes are connected to each other and can exchange data.
+
+Currently, when a node joins/comes up on the network, it relies, by default, on hardcoded bootstrappers for 3 things:
+
+1. initializing its store with the first (genesis) block
+2. requesting the network head to which it sets its initial sync target, and
+3. bootstrapping itself with some other peers on the network which it can utilise for its operations.
+
+This is a centralization bottleneck as it happens both during initial start-up and for any re-starts.
+
+In this ADR, we wish to aleviate this problem by allowing nodes to bootstrap from previously seen peers. This is a simple solution that allows the network alleviate the start-up centralisation bottleneck such that nodes can join the network without the need for hardcoded bootstrappers up re-start
+
+(_Original issue available in references_)
+
+## Decision
+
+Periodically store the addresses of previously seen peers in a key-value database, and allow the node to bootstrap from these peers on start-up, by following the methodology of:
+
+* Approach A.
+
+Caveats: In regards to the storage section of approach A, we decided to factorize the store implementation under `das/store.go` to be used by both approach A and the checkpoint store in `daser`.
```

Comment:
I think this point needs clarification, I should add to the ADR's doc.

What's meant in here is basically factorizing the `Put` and `Load` behaviors that are implemented against the `ds.Datastore` which should cause no coupling whatsoever.

Reply:
On second read, the main benefit of extraction is less code repetition, but such factorization will be impactful for daser for sure, so to avoid impacting DASer, I will mention that we will follow the same pattern only and no factorization will take place. 👍🏻

Reply:
I ended up not following this path in the implementation, and the ADR no longer mentions this "factorizing".

For the record, @renaynay I meant by factorizing, DRYing up the code such that we have one base store implementation to be used by both das and peerstore, but that's a forsaken path now.